automation_tools/__init__.py

```diff
@@ -1467,3 +1471,249 @@ def setenforce(mode):
+    PROJECT_ID:
+        Project ID of an openstack project.
+
+    """
+    username = os.environ.get('USERNAME')
+    if username is None:
+        print('The USERNAME environment variable should be defined.')
+    password = os.environ.get('PASSWORD')
+    if password is None:
+        print('The PASSWORD environment variable should be defined.')
+    auth_url = os.environ.get('AUTH_URL')
+    if auth_url is None:
+        print('The AUTH_URL environment variable should be defined.')
+    project_id = os.environ.get('PROJECT_ID')
+    if project_id is None:
+        print('The PROJECT_ID environment variable should be defined.')
+    with client.Client(
+        username=username,
+        api_key=password,
+        auth_url=auth_url,
+        project_id=project_id
+    ) as openstack_client:
+        openstack_client.authenticate()
+        return openstack_client
+
+
+def create_openstack_instance(instance_name, image_name, flavor_name, ssh_key):
+    """Creates openstack Instance from Image and Assigns a floating IP
+    to instance.
+
+    :param instance_name: A string. Openstack Instance name to create.
+    :param image_name: A string. Openstack image name from which instance
+        to be created.
+    :param flavor_name: A string. Openstack flavor_name for instance.
+        e.g m1.small.
+    :param ssh_key: A string. ssh_key 'name' that required to add
+        into this instance.
+
+    ssh_key should be added to openstack project before running automation.
+    Else the automation will fail.
+
+    The following environment variables affect this command:
+
+    USERNAME:
+        The username of an openstack project to login.
+    PASSWORD:
+        The password of an openstack project to login.
+    AUTH_URL:
+        The authentication url of the project.
+    PROJECT_ID:
+        Project ID of an openstack project.
+
+    """
+    network_name = 'satellite-jenkins'
+    openstack_client = get_OpenstackClient()
+    # Validate ssh_key is added into openstack project
+    openstack_client.keypairs.find(name=ssh_key)
+    image = openstack_client.images.find(name=image_name)
+    flavor = openstack_client.flavors.find(name=flavor_name)
+    network = openstack_client.networks.find(label=network_name)
+    floating_ip = openstack_client.floating_ips.create(
+        openstack_client.floating_ip_pools.list()[0].name
+    )
+    # Create instance from the given parameters
+    instance = openstack_client.servers.create(
+        name=instance_name,
+        image=image.id,
+        flavor=flavor.id,
+        key_name=ssh_key,
+        network=network.id
+    )
+    # Assigning floating ip to instance
+    instance_status = False
+    while instance_status is not True:
+        try:
+            instance.add_floating_ip(floating_ip)
+        except novaclient.exceptions.BadRequest:
+            instance_status = False
+            time.sleep(5)
+        else:
+            instance_status = True
+        continue
```

Comment:
No, here continue is not required. Removing 'continue' as while run until instance_status gets True.